components/forms/TextInput/TextInput.tsx

```diff
@@ -62,7 +62,7 @@ export const TextInput = (props: TextInputProps): React.ReactElement => {
         type={type}
         required={required}
         autoComplete={autoComplete ? autoComplete : "off"}
-        aria-describedby={ariaDescribedBy}
+        aria-describedby={`${meta.error} ${"character-count-message-" + id} ${ariaDescribedBy}`}
```

Comment:
Nice! Have you been able to test it? I just did and it reads it when it comes up but unfortunately does not react to dynamic updates of the error message. Not sure if there is a solution for now but will be looking into it ;)

Reply:
@jeberhardt Found this solution:
```
return (
    <>
      {meta.error ? <ErrorMessage>{meta.error}</ErrorMessage> : null}
      <input
        data-testid="textInput"
        className={classes}
        id={id}
        type={type}
        required={required}
        autoComplete={autoComplete ? autoComplete : "off"}
        aria-describedby={ariaDescribedBy}
        placeholder={placeholder}
        {...field}
        onChange={handleTextInputChange}
      />
      {maxLength && remainingCharacters < maxLength * 0.25 && remainingCharacters >= 0 && (
        <div id={"character-count-message-" + id} role="alert">
          {characterCountMessages.part1} {remainingCharacters} {characterCountMessages.part2}
        </div>
      )}
      {maxLength && remainingCharacters < 0 && (
        <div id={"character-count-message-" + id} className="gc-error-message" role="alert">
          {characterCountMessages.part1Error} {remainingCharacters * -1}{" "}
          {characterCountMessages.part2Error}
        </div>
      )}
    </>
  );
```

It uses `role="alert"` which acts as an ARIA live region and will read dynamic content when updated. The only remaining issue which I am not sure we can do something about is that depending on what you enter in the textfield, VoiceOver will mix the word you just entered with the alert message ("X characters left"). I will let you see by yourself and tell me if you also have the same behavior.

Reply:
I've made an update to it. It will read the updated `describeby` value every time the field gains focus.

Reply:
I had inadvertently used the `meta-error` value, when really I needed to use an ID. That may have been causing it to not read out any values.

Reply:
The spec that we are working from (https://design-system.service.gov.uk/components/character-count/) does not use `role="alert"`.

Reply:
@jeberhardt Just tested your latest changes but it does not work the same way as the UK version. UK version is close to what I would expect even if it has a small flaw (which is probably not something that can be solved). I guess you could compare the two versions and see what we are missing on our side 😕 

Something I noticed with our version is that VoiceOver will only read the updated number of characters (just the number) instead of the the full sentence. Here is a solution I found:
````
<div id={"characterCountMessage" + id} aria-atomic="true">
````

Reply:
Ah, I see now that the UK code uses ` aria-live="polite"` on the character count message. I'll try that, and maybe `aria-atomic` isn't required.